docs/life-of-a-request.md

```diff
@@ -0,0 +1,200 @@
+# Life of a Request
+
+It can sometimes be confusing to figure out how to write a new feature,
+or debug an existing one. Let's try to follow a request through the
+Zulip codebase, and dive deep into how each part works.
+
+## A request is sent to the server, and handled by Nginx
+
+Everything in Zulip goes through nginx, when deployed in production.
+For the most part you don't need to know how this works, except for when
+it isn't working. Nginx does the first level of routing--deciding which
+application will serve the request (or deciding to serve the request
+itself for static content).
+
+## Nginx secures traffic with SSL
+
+If you visit your zulip server in your browser and discover that your
+traffic isn't being properly encrypted, an nginx configuration is the
+likely culprit.
+
+## Static files are served directly by Nginx
+
+Static files include javascript, css, static assets (like emoji, avatars),
+and user uploads (if stored locally and not on S3).
+
+```
+/home/zulip/prod-static
+```
+
+## Nginx routes other requests between tornado and django
+
+All our connected clients hold open long-polling connections so that
+they can recieve events (messages, presence notifications, and so on) in
+real-time. Events are served by Zulip's `tornado` application.
+
+Nearly every other kind of request is served by the `zerver` Django
+application.
+
+# Django routes the request to a view in urls.py files
+
+There are various urls.py files throughout the server codebase, and you
+can read about it more in [the directory structure doc](http://zulip.readthedocs.io/en/latest/directory-structure.html).
+
+The main Zulip Django app is `zerver`. The routes are found in
+```
+zproject/urls.py
+zproject/legacy_urls.py
+```
+
+There are HTML-serving, REST API, legacy, and webhook url patterns.
+
+## Views serving HTML are internationalized by server path
+
+If you look in [zproject/urls.py](https://github.com/zulip/zulip/blob/master/zproject/urls.py), you'll see something called
```

Comment:
It might be worth wrapping this path in backticks as well to maintain consistency: [`zproject/urls.py`](https://github.com/zulip/zulip/blob/master/zproject/urls.py)